### ISSUE TITLE
Keep Git Graph tab pinned after restarting VS Code 

### ISSUE THREAD

**Issue by vinaypatle** [mhutchie/vscode-git-graph#675]
Git Graph tab can be pinned in VS Code which is nice, but when you close VS code and open next time, it is no longer pinned. You have to pin the tab every time you start the VS Code editor. If possible, this would be a nice improvement to have. Thanks

**Comment by borisghidaglia** [comment]
Seems a duplicate of #348.
And as mentioned by @mhutchie [here](https://github.com/mhutchie/vscode-git-graph/issues/348#issuecomment-658729452), you can track the status of this in #276, which has been included in the backlog already.

**Comment by aliehsdev** [comment]
I solved this problem using tasks: when opening VS Code, Git Graph automatically starts and pins.
Add the following tasks for a user (globally) or some workspace:
```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Open Git Graph",
            "command": "${command:git-graph.view}",
            "hide": true,
            "problemMatcher":[],
        },
        {
            "label": "Pin Active Editor",
            "command": "${command:workbench.action.pinEditor}",
            "hide": true,
            "problemMatcher":[],
        },
        {
            "label": "Open and Pin Git Graph",
            "hide": true,
            "dependsOn": [
                "Open Git Graph",
                "Pin Active Editor",
            ],
            "dependsOrder": "sequence",
            "problemMatcher": [],
            "runOptions": {
                "runOn": "folderOpen",
            },
        },
    ]
}
```

To run automatic tasks while the workspace is open, you need to Allow Automatic Tasks (see [Run behavior](https://code.visualstudio.com/docs/editor/tasks#_run-behavior)).

**Comment by borisghidaglia** [comment]
Thanks for sharing!

**Comment by dgilbertsei** [comment]
A recent vscode update must have changed something with the tasks.json format and unfortunately the prior code no longer works.

**Comment by aliehsdev** [comment]
> A recent vscode update must have changed something with the tasks.json format and unfortunately the prior code no longer works.

I'm using the latest version of VS Code, and everything works. Maybe you haven't allowed automatic tasks?

**Comment by dgilbertsei** [comment]
> > A recent vscode update must have changed something with the tasks.json format and unfortunately the prior code no longer works.
> 
> I'm using the latest version of VS Code, and everything works. Maybe you haven't allowed automatic tasks?

Yes, seems to be working again for me. I did have automatic tasks enabled and I hadn't changed anything (afaik) on my side, so not sure why it stopped working or why it's working again now. Maybe I'm just crazy :)

**Comment by pacharanero** [comment]
Is there any possibility this might actually be implemented as a fix? It is annoying that pinning the Git Graph tab works differently (non-persistent across restarts of VS COde) to pinning any other tab (persistent and remain where you left them when you return to a project). I appreciate it's blocked by an underlying VS Code limitation, but the tracking issue https://github.com/mhutchie/vscode-git-graph/issues/276 did say it was roadmapped for fixing, and then that was closed. It's been reported a [number of times](https://github.com/mhutchie/vscode-git-graph/issues/767#issuecomment-1543337740) by a number of users over a number of years, so clearly it's a QOL issue that people would like to see fixed.
